src/Symfony/Component/Routing/Tests/RequestContextTest.php

```diff
@@ -38,6 +41,20 @@ public function testConstruct()
         $this->assertSame(444, $requestContext->getHttpsPort());
         $this->assertEquals('/baz', $requestContext->getPathInfo());
         $this->assertEquals('bar=foobar', $requestContext->getQueryString());
+        $this->assertEquals(['foo' => 'bar'], $requestContext->getParameters());
+    }
+
+    public function testConstructParametersBcLayer()
```

Comment:
not sure what this test case is about

Reply:
its about why the `if` is needed for BC reasons.

Reply:
This lines. As the require for the bc case in this tests, other tests wouldn't catch that.

```diff
+        if ($parameters) {
            $this->parameters = $parameters;
+        }
```

First I wanted to trigger deprecation so we could just remove it with 8.0.